PSICompile/ILCodeGen.cs

```diff
@@ -121,34 +192,48 @@ public class ILCodeGen : Visitor {
    public override void Visit (NUnary u) {
       u.Expr.Accept (this);
       string op = u.Op.Kind.ToString ().ToLower ();
-      op = op switch { "sub" => "neg", _ => op };
+      op = op switch { "sub" => "neg", "not" => "ldc.i4.0\n    ceq", _ => op };

```

Comment:
`WriteLn (not 2)` should print -3 (NOT when applied to an integer should do a simple binary not operator), so the opcode generated for not should depend on the type of operand. This is correctly for boolean.

Reply:
Fixed it by using the **when**  switch expression modifier.